pkg/sync/syncer.go

```diff
@@ -1532,6 +1559,19 @@ func (s *syncer) syncGrantsForResource(ctx context.Context, resourceID *v2.Resou
 
 	resource := resourceResponse.Resource
 
+	shouldSkipGrants, err := s.shouldSkipGrants(ctx, resource)
```

Comment:
syncGrantsForResource is only called by SyncGrants, which has this check, so I think this code can be removed.